src/js/jsx/sections/style/Radius.jsx

```diff
@@ -44,68 +44,137 @@ define(function (require, exports, module) {
+                        return sides.concat(Immutable.List.of(layer.bounds.width / 2, layer.bounds.height / 2));
+                    }, Immutable.List())
+                    .min();
+
+            this.setState({
+                layers: layers,
+                scalars: scalars,
+                maxRadius: maxRadius,
+                maxRadiusInput: Math.floor(maxRadius)
+            });
+        },
+
+        _handleScrubStart: function () {
+            var currentRadius = collection.uniformValue(this.state.scalars);
+
+            // This initial call helps set the history state right before coalescing
+            if (currentRadius !== null) {
+                this.setState({
+                    scrubRadius: currentRadius
+                });
+
+                this.startCoalescing();
+            }
+        },
+
+        _handleRadiusScrub: function (deltaX) {
+            if (this.state.scrubRadius === null) {
+                return;
+            }
+
+            var newRadius = this.state.scrubRadius + deltaX,
+                currentRadius = collection.uniformValue(this.state.scalars);
+
+            newRadius = math.clamp(newRadius, 0, this.state.maxRadius);
+
+            if (newRadius !== currentRadius) {
+                this.getFlux().actions.transform.setRadiusThrottled(
+                    this.props.document,
+                    this.state.layers,
+                    newRadius,
+                    { coalesce: this.shouldCoalesce() }
+                );
+            }
+        },
+
+        _handleScrubEnd: function () {
+            this.setState({
+                scrubRadius: null
+            });
+
+            this.stopCoalescing();
         },
 
         /**
          * Update the radius of the selected layers in response to user input.
          *
-         * @param {Immutable.Iterable.<Layer>} layers
          * @param {SyntheticEvent} event
          * @param {number=} value
          */
-        _handleRadiusChange: function (layers, event, value) {
+        _handleRadiusChange: function (event, value) {
             if (value === undefined) {
                 // In this case, the value is coming from the DOM element
                 value = math.parseNumber(event.target.value);
             }
 
-            var coalesce = this.shouldCoalesce();
+            var options = {
+                coalesce: this.shouldCoalesce()
+            };
+
             this.getFlux().actions.transform
-                .setRadiusThrottled(this.props.document, layers, value, coalesce);
+                .setRadiusThrottled(this.props.document, this.state.layers, value, options);
         },
 
         render: function () {
-            var layers = this.props.document.layers.selected.filter(function (layer) {
-                    return layer.radii;
-                });
-
-            // If there is not at least one selected vector layer, don't render
-            if (layers.isEmpty()) {
+            if (!this.state) {
```

Comment:
Is this still necessary, now that you are calculating state during `componentWillMount`?  Maybe so, I just felt like maybe this was originally introduced because cWRP was not called for initial render...